Server/mods/deathmatch/logic/CModelManager.cpp

```diff
@@ -0,0 +1,39 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *               (Shared logic for modifications)
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        mods/deathmatch/logic/CClientModelManager.cpp
+ *  PURPOSE:     Model manager class
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+
+bool CModelManager::RequestModel(ushort modelID, ushort parentID, eModelType modelType)
+{
+    if (IsValidModel(modelID))
+        return false;
+
+    MapSet(m_ModelsParent, modelID, parentID);
+    MapSet(m_ModelsType, modelID, modelType);
```

Comment:
use operator[]:
```cpp
m_ModelsParent[modelID] = parentID;
m_ModelsType[modelID] = modelType;
```
Altough, I'd recommend creating a structure like `ModelInfo`:
```cpp
struct ModelInfo 
{
unsinged short parentID;
eModelType type;
};
```
And storing it it in the map instead.
Or perhaps since there's a fixed number of models (20000), you could create a static array.